### PR TITLE
Update for Wagtail v3 & v4

### PR DESCRIPTION
**Pull request by nickmoreton**
This makes changes for compatibility with Wagtail 3+ and some changes to add Wagtail v4+ compatibility.

TBX support ticket: https://projects.torchbox.com/projects/support-team/tickets/585

### CODE REVIEW

**Comment by nickmoreton** [comment]
Sorry for the CI noise, I'll work on a fix asap

**Comment by gasman** [comment]
Thanks for looking into this @nickmoreton! On other projects we've found that we need to set POSTGRES_PASSWORD at the same time as bumping the Postgres version past 10: https://github.com/wagtail/wagtail-review/commit/806c55caa94758caeecdf3790ef8e5fa8f3feae4

**Comment by nickmoreton** [comment]
Thanks @gasman for the pointer 👍

**Comment by nickmoreton** [comment]
Hi @gasman The CI is working OK now.

**Comment by nickmoreton** [comment]
https://github.com/torchbox/wagtail-webstories/blob/388bdbc65c1f9b8fae07533ec60bc9fc3477352c/wagtail_webstories/models.py#L58

I spotted a `RemovedInWagtail50Warning` in a site that uses this. It will need updating.

**Comment by nickmoreton** [comment]
The above https://github.com/torchbox/wagtail-webstories/pull/19#issuecomment-1252220770 is now fixed in https://github.com/torchbox/wagtail-webstories/pull/19/commits/ed6d8f502463245d8439c77a581a251e6d6eb3a1

File: tests/models.py [hunk 1]
```diff
@@ -14,14 +25,28 @@ class StoryPage(BaseWebStoryPage):
...
 
 class BlogPage(Page):
-    body = StreamField([
-        ('heading', blocks.CharBlock()),
-        ('story_embed', StoryEmbedBlock()),
-        ('story_link', StoryChooserBlock()),
-        ('external_story_embed', ExternalStoryEmbedBlock()),
-        ('external_story_link', ExternalStoryBlock()),
-    ])
+    if WAGTAIL_VERSION >= (3, 0):
```
Comment (KIRA009) [inline tests/models.py:28]: For some reason, I never thought conditionally defining fields was possible. TIL I guess :)

**Review by KIRA009 — APPROVED** [review]

**Comment by nickmoreton** [comment]
Superseded by https://github.com/torchbox/wagtail-webstories/pull/22

**Pull request closed without merging**